src/overlays/actors/ovl_En_Wiz/z_en_wiz.c

```diff
@@ -30,199 +87,1467 @@ const ActorInit En_Wiz_InitVars = {
+    /* Light arrow    */ DMG_ENTRY(2, EN_WIZ_DMGEFF_LIGHT_ORB),
+    /* Goron spikes   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Deku spin      */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
+    /* Deku bubble    */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
+    /* Deku launch    */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* UNK_DMG_0x12   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_UNK1),
+    /* Zora barrier   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Normal shield  */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Light ray      */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Thrown object  */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
+    /* Zora punch     */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
+    /* Spin attack    */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
+    /* Sword beam     */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Normal Roll    */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* UNK_DMG_0x1B   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* UNK_DMG_0x1C   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Unblockable    */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* UNK_DMG_0x1E   */ DMG_ENTRY(0, EN_WIZ_DMGEFF_IMMUNE),
+    /* Powder Keg     */ DMG_ENTRY(1, EN_WIZ_DMGEFF_NONE),
 };
 
-#endif
+void EnWiz_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnWiz* this = THIS;
+
+    SkelAnime_InitFlex(play, &this->skelAnime, &gWizrobeSkel, &gWizrobeIdleAnim, this->jointTable, this->morphTable,
+                       WIZROBE_LIMB_MAX);
+    SkelAnime_InitFlex(play, &this->ghostSkelAnime, &gWizrobeSkel, &gWizrobeIdleAnim, this->ghostBaseJointTable,
+                       this->ghostMorphTable, WIZROBE_LIMB_MAX);
+    Actor_SetScale(&this->actor, 0.0f);
+    this->platformLightAlpha = 0;
+    this->alpha = 255;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 3;
+    this->unk_450 = 1.0f;
+    this->actor.shape.yOffset = 700.0f;
+    Collider_InitAndSetJntSph(play, &this->ghostColliders, &this->actor, &sJntSphInit, this->ghostColliderElements);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->staffFlameScroll = Rand_S16Offset(0, 7);
+    this->switchFlag = EN_WIZ_GET_SWITCHFLAG(&this->actor);
+    this->type = EN_WIZ_GET_TYPE(&this->actor);
+
+    if (this->switchFlag == 0x7F) {
+        this->switchFlag = -1;
+    }
+
+    if ((this->type == EN_WIZ_TYPE_FIRE) || (this->type == EN_WIZ_TYPE_FIRE_NO_MINI_BOSS_BGM)) {
+        this->actor.colChkInfo.damageTable = &sFireWizrobeDamageTable;
+        this->actor.colChkInfo.health = 8;
+        this->actor.flags &= ~ACTOR_FLAG_100000;
+    } else {
+        this->actor.colChkInfo.damageTable = &sIceWizrobeDamageTable;
+        this->actor.colChkInfo.health = 6;
+    }
+
+    if ((this->switchFlag >= 0) && (Flags_GetSwitch(play, this->switchFlag))) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    this->actor.hintId = TATL_HINT_ID_WIZROBE;
+    this->currentPlatformIndex = 777;
```

Comment:
Guess this is a magic number. Worth a define? Also I think we can probably shorten to `cur` without loosing any information, we do in plenty of other places.

Reply:
This 777 is never used anywhere else; it just seems like an arbitrarily-chosen value to initialize this to. I'd be fine with defining it, but I'm also a bit worried that people will give it more importance that it probably deserves because it has a define; it's literally just "an arbitrary number that's higher than `curPlatformIndex` can ever normally be".

Reply:
I decided to just do it anyways, and you can tell me if it's good or bad-looking. If you don't like it, I'll be happy to revert. If you *do* like it, I'll be happy to keep it.